api/v1alpha4/azurecluster_default_test.go

```diff
@@ -803,6 +803,253 @@ func TestNodeOutboundLBDefaults(t *testing.T) {
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+							{
+								Role:          SubnetNode,
+								Name:          "node-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+						},
+					},
+				},
+			},
+			output: &AzureCluster{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "cluster-test",
+				},
+				Spec: AzureClusterSpec{
+					NetworkSpec: NetworkSpec{
+						Subnets: Subnets{
+							{
+								Role:          SubnetControlPlane,
+								Name:          "control-plane-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+							{
+								Role:          SubnetNode,
+								Name:          "node-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+						},
+						APIServerLB: LoadBalancerSpec{
+							Type: Public,
+						},
+						NodeOutboundLB: &LoadBalancerSpec{
+							Name: "cluster-test",
+							SKU:  SKUStandard,
+							FrontendIPs: []FrontendIP{{
+								Name: "cluster-test-frontEnd",
+								PublicIP: &PublicIPSpec{
+									Name: "pip-cluster-test-node-outbound",
+								},
+							}},
+							Type:                 Public,
+							FrontendIPsCount:     to.Int32Ptr(1),
+							IdleTimeoutInMinutes: to.Int32Ptr(DefaultOutboundRuleIdleTimeoutInMinutes),
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "NAT Gateway enabled - no LB",
+			cluster: &AzureCluster{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "cluster-test",
+				},
+				Spec: AzureClusterSpec{
+					NetworkSpec: NetworkSpec{
+						APIServerLB: LoadBalancerSpec{Type: Public},
+						Subnets: Subnets{
+							{
+								Role:          SubnetControlPlane,
+								Name:          "control-plane-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+							{
+								Role:          SubnetNode,
+								Name:          "node-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+								NatGateway: NatGateway{
+									Name: "node-natgateway",
+								},
+							},
+						},
+					},
+				},
+			},
+			output: &AzureCluster{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "cluster-test",
+				},
+				Spec: AzureClusterSpec{
+					NetworkSpec: NetworkSpec{
+						Subnets: Subnets{
+							{
+								Role:          SubnetControlPlane,
+								Name:          "control-plane-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+							{
+								Role:          SubnetNode,
+								Name:          "node-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+								NatGateway: NatGateway{
+									Name: "node-natgateway",
+								},
+							},
+						},
+						APIServerLB: LoadBalancerSpec{
+							Type: Public,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "NAT Gateway enabled on 1 of 2 node subnets",
```

Comment:
This scenario includes a single node pool without NAT Gateway enabled, so we should expect to get the single, default NodeOutboundLB injected.